integration/app.py

```
@@ -80,88 +89,85 @@ def create_check():
-            city_code=data.get("city_code"),
-            transportation=data.get("transportation"),
-            start_url=data.get("start_url"),
-            external_id=data.get("external_id"),
-            candidate_id=data.get("candidate_id"),
-        )
-        response_data = background_check_adapter.get_check(data=check_data)
-        return response_data
+        check_data = get_check_data(data)
+
+        try:
+            response_data = background_check_adapter.get_check(data=check_data)
+            return response_data
+        except BadRequestAPIException as e:
+            logger.info("BGC adapter bad request exception", extra=get_logger_data(check_data, e.error_message))
+            return get_error_response(e, 400)
+        except ForbiddenAPIException as e:
+            logger.info("BGC adapter forbidden exception", extra=get_logger_data(check_data, e.error_message))
+            return get_error_response(e, 400)
+        except NotFoundAPIException as e:
+            logger.info("BGC adapter not found exception", extra=get_logger_data(check_data, e.error_message))
+            return get_error_response(e, 400)
+        except ServiceUnavailableAPIException as e:
+            logger.info("BGC adapter service unavailable exception", extra=get_logger_data(check_data, e.error_message))
+            return get_error_response(e, 400)
+        except TimeoutAPIException as e:
+            logger.info("BGC adapter timeout exception", extra=get_logger_data(check_data, e.error_message))
+            return get_error_response(e, 400)
+        except UnauthorizedAPIException as e:
+            logger.info("BGC adapter unauthorized exception", extra=get_logger_data(check_data, e.error_message))
+            return get_error_response(e, 400)
+        except UnprocessableEntityAPIException as e:
+            logger.info("BGC adapter unprocessable entity exception", extra=get_logger_data(check_data, e.error_message))
```

We just need to catch `GenericAPIException`. all exceptions raised that extend that one will be caught. The logs could get the message from the exception.  maybe `logger.info("BGC request error {e.error_message}"`

we could also have a decorator for the view that caught `GenericAPIException` and log and return the error message. What do you think @tomasdiaztoro @jeffersonlizar ?

---

agree with the first, the decorator seems a bit unneeded

---

but we said that the Generic exception must reach sentry 👁️

---

We need to raise an exception for not handled responses. we were using `GenericAPIException`. but we could also define another one that does not extend from GenericAPIException.

---

the use of the decorator for the exception handling is optional. I don't have a strong opinion thee